File: 3.ödev/Colony/src/colony/ATaktik.java
/**
*
* @author Sena Nur Erdem     sena.erdem1@ogr.sakarya.edu.tr
* @since 23 Mayıs 2023
* <p>
*  Rastgele bir çift sayı üreten bir taktik sınıfı.
* </p>
*/

package colony;
import java.util.Random;

//ATaktik sınıfı, Taktik arayüzünü uygular
class ATaktik implements Taktik {
    public int savas() {
        Random rand = new Random();
        int rastgele_sayi = (rand.nextInt(501)) * 2; // Rastgele bir çift sayı üret, 0 ile 1000 arasında
        return rastgele_sayi;
    }
}

File: 2.ödev/ColonyClash/src/ATaktik.c
/**
*
* @author Sena Nur Erdem     sena.erdem1@ogr.sakarya.edu.tr
* @since 5 Mayıs 2023
*/

#include "ATaktik.h"
#include <stdlib.h>

int ataktik_savas(Taktik *taktik) {
    int rastgele_sayi = (rand() % 501) * 2; // Rastgele bir çift sayı üret, 0 ile 1000 arasında
    return rastgele_sayi;
}

Taktik *ataktik_olustur() {
    return taktik_olustur(ataktik_savas);
}
